test/CMakeLists.txt

```diff
@@ -79,10 +79,11 @@ endif()
 
 if(MIOPEN_TEST_MIOTENSILE)
     set(SKIP_ALL_EXCEPT_TESTS test_conv2d test_conv3d test_immed_conv2d test_immed_conv3d test_gru test_rnn_vanilla test_lstm test_gru_extra test_rnn_extra test_lstm_extra )
+    list(APPEND SKIP_TESTS test_conv_igemm_dynamic test_conv_igemm_dynamic_small test_conv_for_implicit_gemm)
 endif()
 
 function(add_test_command NAME EXE)
-    if(NOT (NAME IN_LIST SKIP_ALL_EXCEPT_TESTS))
+    if((NOT (NAME IN_LIST SKIP_ALL_EXCEPT_TESTS)) AND (MIOPEN_TEST_INT8 OR MIOPEN_TEST_BFLOAT16 OR MIOPEN_TEST_MIOTENSILE))
```

Comment:
Please add a comment (to the CmakeLists file) that explains why this new condition is needed.

Reply:
It's not new. Please check the original file.
If this condition is missing, all the tests will be skipped because they are not in skip_all_expect_tests list. This condition restrict the usage of this list in certain tests.

Reply:
[don't resolve/hide conversation] Okay, resolved. But this stuff should be eventually rewritten. It is totally puzzling :exploding_head:

Reply:
I can add a comment here.

Reply:
I would highly appreciate that. I still can't figure out how it works;)

Reply:
done